### DISCUSSION TITLE
[Android] Remove non-existing/non-3rd party/signature/priviledged permissions from the Android export dialog

### DISCUSSION THREAD

**Discussion by melquiadess** [godotengine/godot-proposals#9492] [Platforms]
In the Android export window (and so in `EditorExportPlatformAndroid.xml` file), in the `Permissions` section, , there are:

- permissions that were removed from Android API, eg. `ACCESS_SURFACE_FLINGER`, `READ_HISTORY_BOOKMARKS`, ... - and they are not present in the Android documentation anymore
- permissions that are marked **"Not for use by third-party applications."** in the documentation, eg. `ACCESS_CHECKIN_PROPERTIES` 
- permissions with `Protection level` being a combination of: `signature`, `privileged`, `development` - effectively out of bounds for normal devs

![Screenshot 2024-04-09 180128](https://github.com/godotengine/godot-proposals/assets/13484849/6eb11a9f-e375-4b37-99e3-211ee3546e97)

![Screenshot 2024-04-09 180414](https://github.com/godotengine/godot-proposals/assets/13484849/029f8d1d-dbbd-415a-a4c2-0aae22993a85)

### Is there a use case for any of those in Godot? 

It seems to me that there isn't, so unless I'm missing something, I would like to ~kill'em all~ remove them.

We could also keep them, and use `deprecated` to inform, eg.

```
<member name="permissions/write_profile" type="bool" setter="" getter="" deprecated="Removed from Android API">
	Allows an application to write (but not read) the user's personal profile data.
</member>
```

but I don't see the point of maintaining something that's gone from the Android API/not available to 3rd party devs.

**Comment by Calinou** [comment]
> Is there a use case for any of those in Godot?

It's technically possible to use Godot to develop an app that targets rooted devices only, but it's extremely unlikely, especially in this day and age where rooting is much less common than it used to be.

**Comment by melquiadess** [comment]
> > Is there a use case for any of those in Godot?
> 
> It's technically possible to use Godot to develop an app that targets rooted devices only, but it's extremely unlikely, especially in this day and age where rooting is much less common than it used to be.

Thanks @Calinou 

- permissions removed from the API (**28** of them still present in Export dialog) should never be used anymore, unless people want to target very low API levels (eg. many of those permissions were removed in API 23), which seems unlikely
- permissions marked "Not for use by third-party" - (**37**) should probably also not be used 
- permissions with protection level "signature" - (**~13**) these mean that the app must be signed with the same certificate as the OS, which implicitly could mean rooted devices with a custom rom (with the caveat that the device being rooted is not enough in itself, it's the certificates that must match), but I agree, this is also unlikely

I think the use case for those permissions is weak/non-existent, and if required, they could be added in the `custom permissions` array.

It would mean we can get rid of ~**80** permissions from the codebase.

**Reply by melquiadess** [reply]
###### On the other hand, there are ~150 _new_ permissions in Android, but not present in the export dialog. I don't believe all of them should be added, just some of the most common ones, but that's for another PR/discussion.

**Comment by melquiadess** [comment]
This has been discussed in https://github.com/godotengine/godot/pull/90547 and another approach has been proposed. Closing...
